### DISCUSSION TITLE
Issue with MainMenu Customization

### DISCUSSION THREAD

**Discussion by Digisaman** [OrchardCMS/OrchardCore#13408] [General]
I have an issue with navigation, I'm trying to change the MainMenu Items based on  the logged-in user's Role ( add menu items specific to a role) I've tried 2 approaches 
1 : using INavigationProvider in theme to dynamically add Menu Items based on Role, 
2: Adding the menu Items in admin dashboard and then decide not to render some menus based on role in Menu.cshtml in theme. 

Both Approaches work correctly in dev environment(Visual Studio 2022), but When I publish to my local IIS, Or the remote WebServer, it doesn't work!! I've been stuck on this issue for weeks, please help me

**Comment by hishamco** [comment]
I'm wondering why it doesn't work remotely!! How you customize your navigation? Are you using recipes?
